privacyidea/lib/smsprovider/SmppSMSProvider.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+#
+#    E-mail: yurlov.alexandr@gmail.com
+#
+#    2017-12-22 Alexander Yurlov <yurlov.alexandr@gmail.com>
+#               Add new SMPP SMS provider
+
+
+__doc__="""This is the SMSClass to send SMS via SMPP protocol to SMS center
+It requires smpplib installation, this lib works with ascii only, but message support unicode 
+
+"""
+
+from privacyidea.lib.smsprovider.SMSProvider import (ISMSProvider, SMSError)
+from privacyidea.lib import _
+import requests
+from urlparse import urlparse
+
+import logging
+log = logging.getLogger(__name__)
+
+import smpplib
+
+class SmppSMSProvider(ISMSProvider):
+
+    def submit_message(self, phone, message):
+        """
+        send a message to a phone via an smpp protocol to smsc
+
+        :param phone: the phone number
+        :param message: the message to submit to the phone
+        :return:
+        """
+        log.debug("submitting message {0!s} to {1!s}".format(message, phone))
+        parameter = {}
+        if self.smsgateway:
+            smsc_host = self.smsgateway.option_dict.get("SMSC_HOST")
+            smsc_port = self.smsgateway.option_dict.get("SMSC_PORT")
+            sys_id = self.smsgateway.option_dict.get("SYSTEM_ID")
+            passwd = self.smsgateway.option_dict.get("PASSWORD")
+            s_addr_ton = int(self.smsgateway.option_dict.get("S_ADDR_TON"),16)
```

Comment:
Are there any default values for TON and NPI? Otherwise we also need to fail, if they are not present!

Reply:
@yurlov-alexander according to https://docs.aerialink.net/api/smpp/ton-npi-settings/ I would default to ``0``.
Is it really common practice to provide these number as hex and not as base10?

Reply:
I've seen my own configuration. There are the following parameters:
D_ADDR_NPI= 0x01
D_ADDR_TON= 0x01
S_ADDR_NPI= 0x00
S_ADDR_TON= 0x05
These parameters depend from sms center, so I can't predict default configuration.